client/my-sites/checkout/checkout-thank-you/pending/index.tsx

```diff
@@ -237,6 +248,9 @@ function useRedirectOnTransactionSuccess( {
 		}
 
 		didRedirect.current = true;
+		if ( isConnectAfterCheckoutFlow ) {
+			setHeadingText( connectingJetpackText );
+		}
```

Comment:
Something that kinda bugs me is that this message is displayed, and right after that the connection screen appears. While when I'm reading the message, it makes me think that the connection is happening right at that time.

Reply:
Thanks for reviewing and for your feedback @IanRamosC!
Hmm, yes I see what you're saying, but the connection _is_ happening right at that time, - it's the "**site** connection" that is taking place. Then after the site connection, you see the authorization/connection screen. It's asking you if you then want to approve the "**user** connection". Once you click approve, then the "**user** connection" then takes place.
I thought it was a good idea to show this message when the "site" connection is taking place because sometimes the site connection can take a considerable amount of time (for whatever reason- server load is high, user's internet connection is slow, etc). And similarly, the transaction process can also sometimes take a considerable amount of time (server load is high, user connection is slow, some payment processors take longer than others such as PayPal). So therefore, if it just so happens that the users transaction takes longer than usual, and then also the site connection takes longer than usual, then if we're only showing the one message during this whole process ("Almost there – we're currently finalizing your order.") it will show for a good long time and can appear to the user that something is frozen or stuck. In fact that is what happened to James Grierson (except the connection process was taking place "before" checkout) and this is what prompted him to request this project in the first place. Adding this additional message telling the user that the transaction has finalized, but now we're connecting Jetpack, I feel it better communicates to the user that nothing is frozen or stuck, and it properly let's the user know what is taking place. But if you feel this could be improved somehow or better communicated, please do let me know your thoughts! 🙂

Reply:
Thanks for the explanation, that is a really good point.